### DISCUSSION TITLE
How do I change or disable the ceph dashboard auto logout time?

### DISCUSSION THREAD

**Discussion by ftab** [rook/rook#10203] [Q&A] [answered]
The dashboard logs me out in a matter of minutes if I'm not actively messing around in it jumping between pages and whatnot. It would be nice to extend or even remove that automatic timeout so that it's easier for me to maintain "battle stations" tabs in the browser to keep tabs on my cluster.

I have extended the `--token-ttl` on my kubernetes dashboard for the same reason, but I can't find a similar setting in `ceph dashboard --help` or in the docs here https://docs.ceph.com/en/quincy/mgr/dashboard/ - I've tried looking for token, ttl, session, timeout, logout, length, and seconds, to no avail.

**Comment by travisn** [comment]
@epuertat Is there a dashboard setting for the auto logout timeout?

**Comment by epuertat** [comment, marked as answer]
Hi @travisn @ftab,

The only (intentional) mechanism to expire sessions is the JWT TTL (`mgr/dashboard/jwt_token_ttl` accesible via `ceph config [set|get]` or via `ceph dashboard [set|get]-jwt-token-ttl`). However, the default setting for that is 24 hours (in seconds).

Nonetheless, what I've observed is that when running multiple Ceph-Dashboards (Ceph clusters actually) behind the same IP/hostname (via reverse proxy) and from the same browser (or mixing "admin" and "guest" users at the same time), when I log in with one account/to one cluster, it logs me off from the other sessions. That's basically because the Ceph Dashboard sessions are persisted in the browser, not in the back-end, so the browser cannot have more than 1 session per browser/domain/user account.

@ftab is this is the case, could you please try using different browsers/incognito mode?

**Reply by ftab** [reply]
I'm buried in so much work my head is spinning. Finally got back around to this tab.

```
[rook@rook-ceph-tools-86dbcb7664-tnrg5 /]$ ceph dashboard get-jwt-token-ttl
300
[rook@rook-ceph-tools-86dbcb7664-tnrg5 /]$ ceph dashboard set-jwt-token-ttl 86400
JWT token TTL updated
```

I think that does the trick. Thank you!

Must've been some kind of weird old default (this cluster has been up since around rook 1.3.x). Don't know how I missed it on my first look around.

**Comment by alphabet5** [comment]
So I noticed frequent logouts when trying to provision a new cluster.

The operator seems to be trying to reconcile the cephcluster, failing to enable stretch mode (unbalanced zones), and trying to reconcile again.

```
2025-06-12 19:51:02.321546 I | op-mgr: start running mgr
```

Each cycle it causes any dashboard sessions to be logged out. - I don't think it's an issue, and it's not the token expiring, just another potential cause for a getting logged out frequently.
